File: Variable System/C#/Source/SettingsManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Variable_System {

	public class SettingsManager {

		private static string m_defaultFileName = "settings.ini";

		private VariableSystem m_variables;

		public SettingsManager() {
			m_variables = new VariableSystem();
			reset();
		}

		public void reset() {

		}

		public static string defaultFileName {
			get { return m_defaultFileName; }
		}

		public bool load() { return loadFrom(m_defaultFileName); }

		public bool save() { return saveTo(m_defaultFileName); }

		// load game settings from a specified file name
		public bool loadFrom(string fileName) {
			// use a variable system to parse the settings file
			VariableSystem newVariables = VariableSystem.readFrom(fileName);
			if(newVariables == null) { return false; }

			m_variables = newVariables;

			// create local variables instantiated with data parsed from the variable system
			// TODO

			return true;
		}

		public bool saveTo(string fileName) {
			// update the variable system with the new game settings values
			// TODO

			// group the variables by categories
			m_variables.sort();

			// update the settings file with the changes
			return m_variables.writeTo(fileName);
		}

	}

}

File: Variable System/Java/src/SettingsManager.java
public class SettingsManager {
	
	private VariableSystem m_settings;
	
	final public static String defaultSettingsFileName = "settings.ini";
	
	public SettingsManager() {
		m_settings = new VariableSystem();
		reset();
	}
	
	public void reset() {
		
	}
	
	public boolean load() { return loadFrom(defaultSettingsFileName); }
	
	public boolean save() { return saveTo(defaultSettingsFileName); }
	
	public boolean loadFrom(String fileName) {
		VariableSystem variables = VariableSystem.readFrom(fileName);
		if(variables == null) {
			System.out.println("ERROR: Unable to load settings file: \"" + fileName + "\".");
			return false;
		}
		
		m_settings = variables;
		
		// create local variables instantiated with data parsed from the variable system
		// TODO
		
		return true;
	}
	
	public boolean saveTo(String fileName) {
		// update the variable system with the new settings values
		// TODO
		
		// group the settings by categories
		m_settings.sort();
		
		return m_settings.writeTo(fileName);
	}
	
}
